articles/quickstart.md

```diff
@@ -1,163 +1,129 @@
+   * Right click on the `Operation.qs` file
+   * Rename it to `Bell.qs`
 
-#### Source files for a quantum application
-Your development environment should have two files open:
-`Driver.cs`, which will hold the C# driver for your quantum code,
-and `Operations.qs`, which will hold the quantum code itself.
+* * *
 
-#### Q# source
-The first step is to rename the Q# file to `Bell.qs`.
-Right-click on `Operations.qs` in the Visual Studio Solution Explorer (Ctrl+Alt+L to focus) or the Visual Studio Code Explorer (Ctrl/⌘+Shift+E to focus), and select the Rename option.
-Replace `Operations` with `Bell` and hit return.
+## Write a Q# operation
 
-To enter the Q# code, make sure that you are editing the
-`Bell.qs` window.
-The `Bell.qs` file should have the following contents:
+Our goal is to create a [Bell State](https://en.wikipedia.org/wiki/Bell_state) showing entanglement. We will build this up piece by piece to show the concepts of qubit state, gates, and measurement.
 
-```qsharp
-namespace Quantum.Bell {
-    open Microsoft.Quantum.Intrinsic;
-    open Microsoft.Quantum.Canon;
+### Q# operation code
+
+1. Replace the contents of the Bell.qs file with the following code:
+
+    ```qsharp
+    namespace Quantum.Bell {
+        open Microsoft.Quantum.Intrinsic;
+        open Microsoft.Quantum.Canon;
 
-    operation HelloQ () : Unit {
-        Message("Hello quantum world!");
+        operation Set (desired: Result, q1: Qubit) : Unit {
+            if (desired != M(q1)) {
+                X(q1);
+            }
+        }
     }
-}
-```
+    ```
 
-#### Q# operation
-First, replace the string `HelloQ` with `Set`, and change the operation
-parameters (the content of the parentheses) to contain the string
-`desired: Result, q1: Qubit`.
-The file should now look like:
+    This operation may now be called to set a qubit to a known state (`Zero` or `One`).
 
-```qsharp
-namespace Quantum.Bell {
-    open Microsoft.Quantum.Intrinsic;
-    open Microsoft.Quantum.Canon;
+    We measure the qubit, if it's in the state we want, we leave it alone, otherwise, we flip it with the `X` gate.
 
-    operation Set (desired: Result, q1: Qubit) : Unit {
-        Message("Hello quantum world!");
-    }
-}
-```
+### About Q# operations
 
-Now, replace `Message("Hello quantum world!");` with the following code between the braces that enclose the operation body:
+An operation is the basic unit of quantum execution in Q#. It is roughly equivalent to a function in C or C++ or Python, or a static method in C# or Java.
 
-```qsharp
-        if (desired != M(q1)) {
-            X(q1);
-        }
-```
+The arguments to an operation are specified as a tuple, within parentheses.
 
-The file should now look like:
+The return type of the operation is specified after a colon. In this case, the `Set` operation has no return, so it is marked as returning `Unit`. This is the Q# equivalent of `unit` in F#, which is roughly analogous to `void` in C#.
```

Comment:
Would it be helpful to note that the analogous concept in Python is the empty tuple `()` of type `Tuple[()]`?

| | Value | Type |
|---|---|---|
| Q# | `()` | `Unit` |
| F# | `()` | `unit` | 
| Python | `()` | `Tuple[()]` |